File: BOCA/AI.c
#include <stdio.h>

int main() {
    int grau, q;
    float ai[7], x, res[1000000];

    scanf("%d", &grau);
    for (int i = 0; i < grau+1; i++)
    {
        scanf("%f", &ai[i]);
    }

    scanf("%d", &q);

    for (int i = 0; i < q; i++)
    {
        scanf("%f", &x);
        float soma = 0, pot;
        for (int j = 0; j < grau+1; j++)
        {
            pot = 1;
            for (int p = 0; p < j; p++)
            {
                pot *= x;
            }
            soma+= pot*ai[j];
        }
        res[i] = soma;
    }

    for (int i = 0; i < q; i++)
    {
        printf("%f\n", res[i]);
    }

    return 0;
}

File: Listas/ListaPC1 - Vetores/AI.py
x = []
grau = int(input())
ai = list(map(float, input().split())) 
q = int(input())
for i in range(0, q):
    x.append(float(input()))
for i in x:
    soma = 0.0
    for j in range(grau+1):
        pot = 1.0
        for k in range(0, j):
            pot *= i

        soma+=pot*ai[j]
    print(soma)
